msnc/model.py

```diff
@@ -94,14 +94,13 @@ def run_training(
             model_file_name = '{:04d}.model'.format(epoch)
             self._save(model_file_name)
 
-            if development_set is not None:
-                dev_accuracy = self.run_evaluation(development_set)
-                self._write_log(dev_accuracy, epoch)
-                if dev_accuracy > best_dev_accuracy:
-                    best_dev_accuracy, best_epoch = dev_accuracy, epoch
-                    self._write_log(dev_accuracy, epoch, '[new best]')
-                    if save_best_model:
-                        self._save('best.model')
+            dev_accuracy = self.run_evaluation(development_set)
+            self._write_log(dev_accuracy, epoch)
+            if dev_accuracy > best_dev_accuracy:
+                best_dev_accuracy, best_epoch = dev_accuracy, epoch
+                self._write_log(dev_accuracy, epoch, '[new best]')
+                if save_best_model:
+                    self._save('best.model')
 
         self._write_log(best_dev_accuracy, best_epoch, '[best]')
```

Comment:
When `development_set` is not given, this code wrongly logs `best_epoch` is 0.

Reply:
Nice catch! I didn't realize it.